process/block/interceptors/headerInterceptor.go

```diff
@@ -34,38 +38,81 @@ func NewHeaderInterceptor(
 	chronologyValidator process.ChronologyValidator,
 ) (*HeaderInterceptor, error) {
 
+	if marshalizer == nil {
+		return nil, process.ErrNilMarshalizer
+	}
 	if headersNonces == nil {
 		return nil, process.ErrNilHeadersNoncesDataPool
 	}
 	if headers == nil {
 		return nil, process.ErrNilHeadersDataPool
 	}
-	hdrBaseInterceptor, err := NewHeaderInterceptorBase(
-		marshalizer,
-		storer,
-		multiSigVerifier,
-		hasher,
-		shardCoordinator,
-		chronologyValidator,
-	)
+	if storer == nil {
+		return nil, process.ErrNilHeadersStorage
+	}
+	if multiSigVerifier == nil {
+		return nil, process.ErrNilMultiSigVerifier
+	}
+	if hasher == nil {
+		return nil, process.ErrNilHasher
+	}
+	if shardCoordinator == nil {
+		return nil, process.ErrNilShardCoordinator
+	}
+	if chronologyValidator == nil {
+		return nil, process.ErrNilChronologyValidator
+	}
+
+	hdrInterceptor := &HeaderInterceptor{
+		marshalizer:         marshalizer,
+		storer:              storer,
+		multiSigVerifier:    multiSigVerifier,
+		hasher:              hasher,
+		shardCoordinator:    shardCoordinator,
+		chronologyValidator: chronologyValidator,
+		headers:             headers,
+		headersNonces:       headersNonces,
+	}
+
+	return hdrInterceptor, nil
+}
+
+// ParseReceivedMessage will transform the received p2p.Message in an InterceptedHeader.
+// If the header hash is present in storage it will output an error
+func (hi *HeaderInterceptor) ParseReceivedMessage(message p2p.MessageP2P) (*block.InterceptedHeader, error) {
```

Comment:
improve code coverage, copy test functions from shard and base test to headerInterceptor_test to test every bit.

Reply:
Damn. Forgot that the tests needed merging as well.

Reply:
Did the merging. Now `headerInterceptor_test.go` contains the union of all the tests for these 3 structs, but without any duplicates. I'm not entirely sure that I unified the tests correctly - I don't understand the `headerInterceptor` 100%.

Reply:
Tests seems fine, should add more tests for the contructor

Reply:
Added extra tests for the constructor.